### PR TITLE
Testcontainers integration

### PR DESCRIPTION
**Pull request by auriium**
the fucking modularization keeps breaking it thinks that the sources don't exist even when they do

Theyre being generated on phase generate-sources and are being put into the folder generate sources what else does it even want

Besides these fucking issues are more issues in that i'm eventually going to have to compeltely tear out and replace testcontainers with docker (which a248's pc doesn't even support right now) because testcontainers is not built to be used like this apparently and has a bunch of fucking bullshitty crap that it does ugh

In conclusion this has been a waste of time, i doubt this will ever get merged or leave draft state, and we're probably going to move back to mariadb-maven-plugin because i have no idea what i'm doing or why i'm on the solar mc team

### CODE REVIEW

**Comment by auriium** [comment]
Side note: I've tested it on my discord bot just now. It works fine asides from two points

1: Testcontainers annoying thread bs that i had to fix with the copy pasted crap from the thread that abuses reflection
(solution: docker rewrite never)

2: Modularization does not work, from what i've tested, it always generates the sources into the correct target in the right package, but when reaching the compile phase it complains that they don't exist when they legit do
(solution: no clue)
